applications: nrf_desktop: Module documentation improvements (#10369)
nrfconnect/sdk-nrf: applications/nrf_desktop/doc/hids.rst

The HID Service application module initializes and configures the |GATT_HID|.
The application module registers the HID report map and every HID report that was enabled in the application configuration.

> Detailed information about HID-related configuration in nRF Desktop is available in the :ref:`nrf_desktop_hid_configuration` documentation.

Reviewer: ```suggestion
For detailed information about the HID-related configuration in the nRF Desktop, see the :ref:`nrf_desktop_hid_configuration` documentation.
```

The module can receive an HID output report setting state of the keyboard LEDs, e.g. state of the Caps Lock.
The report is received from the Bluetooth connected host.

> The module forwards the report using ``hid_report_event``, that is handled by |hid_state|.

Reviewer: ```suggestion
The module forwards the report using ``hid_report_event`` that is handled by |hid_state|.
```
FYI: I removed all the commas used in similiar context as these are restrictive clauses.
